core/array.cpp

```diff
@@ -355,6 +355,48 @@ Variant Array::pop_front() {
 	return Variant();
 }
 
+Array *Array::reduce(Object *p_obj, const StringName &p_function, const Variant &p_args) {
```

Comment:
`reduce` should result in a `Variant`, not an array.

The whole implementation is `filter` and not `reduce`.

Also I personally would prefer if you'd use `FuncRef` instead, that might make a future lambda integration a lot easier, but maybe I'm wrong. I am not perfectly up to date with the GDScript roadmap.